tests/5.0/task/test_task_detach.c

```diff
@@ -18,42 +22,50 @@
 
 #define N 1024
 
+int test_callback(omp_event_handle_t event) {
+  omp_fulfill_event(event);
+  return 1;
+}
+
 int test_task_detach() {
   OMPVV_INFOMSG("test_task_detach");
   int errors = 0, x = 0, y = 0;
   int num_threads = -1, record_x = -1, record_y = -1;
-  omp_event_handle_t *flag_event;
+  omp_event_handle_t flag_event;
 
-#pragma omp parallel num_threads(2) default(shared)
+#pragma omp parallel
+#pragma omp single
   {
-    if (omp_get_thread_num() == 1 || omp_get_num_threads() < 2) {
-      num_threads = omp_get_num_threads();
-      x++;
-      omp_fulfill_event(flag_event);
+#pragma omp task depend(out: y) detach(flag_event)
+    {
+      y++;
     }
-    if (omp_get_thread_num() == 0) {
-#pragma omp task detach(flag_event)
-      {
-        y++;
-      }
-#pragma omp taskwait
+#pragma omp task
+    {
+      x = 1;
+#pragma omp flush
+      test_callback(flag_event);
+    }
+#pragma omp task depend(inout: y)
+    {
+#pragma omp flush
       record_x = x;
       record_y = y;
+      num_threads = omp_get_num_threads();
     }
   }
 
   OMPVV_ERROR_IF(num_threads < 0, "Test ran with invalid number of teams (less than zero)");
   OMPVV_WARNING_IF(num_threads == 1, "Test ran with one thread, so the results are not conclusive");
 
-  OMPVV_TEST_AND_SET_VERBOSE(errors, record_x != 1 || record_y != 1);
-  OMPVV_ERROR_IF(record_x == 0, "Taskwait did not wait for associated event to be fulfilled");
-  OMPVV_ERROR_IF(record_y == 0, "Taskwait did not wait for task body to execute");
-  OMPVV_ERROR_IF(record_x == -1 || record_y == -1, "Recording variables were not set in the parallel region as expected.")
+  OMPVV_TEST_AND_SET_VERBOSE(errors, record_x != 1);
+  OMPVV_ERROR_IF(record_x == 0, "Depend did not wait for associated event to be fulfilled");
+  OMPVV_ERROR_IF(record_y == 0, "Depend did not wait for task body to execute");
+  OMPVV_ERROR_IF(record_x == -1 || record_y == -1, "Recording variables were not set correctly")
```

Comment:
Change messages

Reply:
I will wait for the msgs to be updated before approving the request. Also, I don't particularly like clubbing depend and detach together in 64. It is more informative if it is split up.

Reply:
Updated the messages and split the message for unset recording variables on line 64.